src/Shared/FileUtilities.cs

```diff
@@ -836,5 +837,36 @@ internal static string CombinePaths(string root, params string[] paths)
 
             return paths.Aggregate(root, Path.Combine);
         }
+
+        internal static string TrimTrailingSlashes(this string s)
+        {
+            return s.TrimEnd('/', '\\');
+        }
+
+        /// <summary>
+        /// Replace all backward slashes to forward slashes
+        /// </summary>
+        internal static string ToSlash(this string s)
+        {
+            return s.Replace('\\', '/');
+        }
+
+        internal static string NormalizeForPathComparison(this string s) => s.ToSlash().TrimTrailingSlashes();
+
+        internal static bool PathsEqual(string path1, string path2)
```

Comment:
It potentially gets called a lot. Should I hold off the PR until I do some measurements ~next week?

One alternative would be to convert the string to byte arrays and work on those, in-place. But you'd still have 2 new arrays for each call. Another alternative is to rewrite the file matcher from scratch and try to work around the need to do this equality